### PR TITLE
Simplify enabling CPU offload in FSDP

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

A UX improvement for enabling offloading without needing to import extras from the FSDP library.

Before:

```py
from torch.distributed.fsdp.fully_sharded_data_parallel import CPUOffload

strategy = FSDPStrategy(cpu_offload=CPUOffload(offload_params=True))
```

After:
```py
strategy = FSDPStrategy(cpu_offload=True)
```

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @justusschock @awaelchli @carmocca

### CODE REVIEW

File: src/lightning_lite/strategies/fsdp.py [hunk 1]
```diff
@@ -91,7 +90,7 @@ def __init__(
         precision: Optional[Precision] = None,
         process_group_backend: Optional[str] = None,
         timeout: Optional[timedelta] = default_pg_timeout,
-        cpu_offload: Optional["CPUOffload"] = None,
+        cpu_offload: Optional[Union[bool, "CPUOffload"]] = None,
```
Comment (Borda) [inline src/lightning_lite/strategies/fsdp.py:93]:
```suggestion
        cpu_offload: Union[bool, "CPUOffload", None] = None,
```

File: src/pytorch_lightning/strategies/fully_sharded_native.py [hunk 2]
```diff
@@ -115,7 +111,7 @@ def __init__(
         checkpoint_io: Optional[CheckpointIO] = None,
         precision_plugin: Optional[PrecisionPlugin] = None,
         process_group_backend: Optional[str] = None,
-        cpu_offload: Optional[CPUOffload] = None,
+        cpu_offload: Optional[Union[bool, "CPUOffload"]] = None,
```
Comment (Borda) [inline src/pytorch_lightning/strategies/fully_sharded_native.py:114]:
```suggestion
        cpu_offload: Union[bool, "CPUOffload", None] = None,
```
less brackets are easier to read... :rabbit:

**Review by Borda — APPROVED** [review]

File: src/lightning_lite/strategies/fsdp.py (hunk 1 above)
Comment (awaelchli) [reply src/lightning_lite/strategies/fsdp.py:93]: Sure, I'll make the change, but it's not the same style we have in the code base

**Review by justusschock — APPROVED** [review]

File: src/lightning_lite/strategies/fsdp.py [hunk 3]
```diff
@@ -258,7 +257,6 @@ def broadcast(self, obj: TBroadcast, src: int = 0) -> TBroadcast:
     def register_strategies(cls, strategy_registry: Dict) -> None:
         if not _TORCH_GREATER_EQUAL_1_12 or not torch.distributed.is_available():
             return
-        from torch.distributed.fsdp.fully_sharded_data_parallel import CPUOffload
```
Comment (carmocca) [inline src/lightning_lite/strategies/fsdp.py:261]: After this change, when would the user want to instantiate and pass this class? What other arguments does it expose?

Comment (awaelchli) [reply src/lightning_lite/strategies/fsdp.py:261]: > After this change, when would the user want to instantiate and pass this class?

Probably never, but it was the only way so far. It depends on the PyTorch team to decide whether they see other useful arguments to be included in the dataclass or not. My guess is not, they will probably eventually go with a bool (after all, the activation checkpointing has moved to a separate API as well).

**Pull request merged**